OKD: ironic-static-ip-manager OCP image can be used for OKD as well

sherine-k opened the pull request:
This PR is a partial replacement for https://github.com/openshift/release/pull/34166 for ironic-static-ip-manager

LorbusChris commented on core-services/release-controller/_releases/release-ocp-4.13-ci.json:
```diff
@@ -154,4 +153,4 @@
   }
-}
\ No newline at end of file
+}
```
Not sure whether Prow will bark at this

LorbusChris reviewed (approved):
/lgtm

sherine-k commented:
/pj-rehearse

LorbusChris commented:
/assign @derekhiggins @jupierce

elfosardo reviewed (changes requested):
I strongly suggest to use stream9, all the packages used come from cs9/rhel9 and this could break without notice
ironic upstream has removed support for python3.6 and CS8 since the beginning of the year, we removed support in the ironic openshift containers right after

elfosardo commented:
/lgtm
sorry removed the label by mistake, this is good to go

derekhiggins commented:
@sherine-k would you mind posting a little context on whats going on here, I'm not familiar with what your doing.

sherine-k commented:
> @sherine-k would you mind posting a little context on whats going on here, I'm not familiar with what your doing.

Hello @derekhiggins , yes sorry, of course!
Right now, https://origin-release.ci.openshift.org/#4.13.0-0.okd-scos shows that all builds of OKD on SCOS are failing: cluster-bare-metal-operator is not finding the tags for most of the necessary ironic-* images.

Previously we replaced ironic-* images in 4.6+ as we didn't have baremetal installer. We started working on Fedora-based `*.okd` images, but then in 4.10 we rebased on upstream installer and could reuse existing images. This is the case for all ironic* images, if I'm not mistaken, except for `ironic-image` and `ironic-agent-image`.

In this PR, we are removing `__okd ` for ironic* images if we use the same Dockerfile as for OCP.

derekhiggins commented:
/approve
thanks @sherine-k

sherine-k commented:
Hi @derekhiggins 
Do you think we can merge this PR?

elfosardo commented:
/approve

jupierce commented:
/approve
